src/CoreWCF.BuildTools/tests/OperationInvokerGeneratorTests.cs

```diff
@@ -0,0 +1,1046 @@
+        }}
+    }}
+}}
+"
+                    },
+                    AnalyzerConfigFiles =
+                    {
+                        (typeof(OperationInvokerGenerator),"/.globalconfig", """
+is_global = true
+build_property.EnableCoreWCFOperationInvokerGenerator = true
+""")
+                    },
+                    GeneratedSources =
+                    {
+                        (typeof(OperationInvokerGenerator), "OperationInvoker.g.cs", SourceText.From($$"""
+// <auto-generated>
+// Generated by the CoreWCF.BuildTools.OperationInvokerGenerator source generator. DO NOT EDIT!
+// </auto-generated>
+#nullable disable
+using System;
+using System.Threading.Tasks;
+namespace System.Runtime.CompilerServices
+{
+    [AttributeUsage(AttributeTargets.Method, AllowMultiple = false)]
+    file sealed class ModuleInitializerAttribute : Attribute { }
+}
+namespace CoreWCF.Dispatcher
+{
+    // This class is used to invoke the method MyProject.IIdentityService.Echo(System.Collections.Generic.List<string>).
+    file sealed class OperationInvoker0 : CoreWCF.Dispatcher.IOperationInvoker
+    {
+        public ValueTask<(object returnValue, object[] outputs)> InvokeAsync(object instance, object[] inputs)
+        {
+            System.Collections.Generic.List<string> p0;
+            p0 = (System.Collections.Generic.List<string>)inputs[0];
+            var result = ((MyProject.IIdentityService)instance).Echo(p0);
+            var outputs = AllocateOutputs();
+            return new ValueTask<(object, object[])>((result, outputs));
+        }
+
+        public object[] AllocateInputs() => new object[1];
+
+        private object[] AllocateOutputs() => Array.Empty<object>();
+
+        internal static void RegisterOperationInvoker() => CoreWCF.Dispatcher.DispatchOperationRuntimeHelpers.RegisterOperationInvoker("MyProject.IIdentityService.Echo(System.Collections.Generic.List<string>)", new OperationInvoker0());
+    }
+}
+namespace CoreWCF.Dispatcher
+{
+    file sealed class OperationInvokerModuleInitializer
+    {
+        [System.Runtime.CompilerServices.ModuleInitializer]
+        internal static void RegisterOperationInvokers()
+        {
+            OperationInvoker0.RegisterOperationInvoker();
+        }
+    }
+}
+#nullable restore
+
+""", Encoding.UTF8, SourceHashAlgorithm.Sha256)),
+                    },
+                },
+            };
+
+            await test.RunAsync();
+        }
+
+        [Theory]
+        [MemberData(nameof(GetTestVariations))]
+        public async Task ParamsArrayTest(string attributeNamespace)
+        {
+            var test = new VerifyGenerator.Test
+            {
+                TestState =
+                {
+                    Sources =
+                    {
+@$"
+namespace MyProject
+{{
+    [{attributeNamespace}.ServiceContract]
+    public interface IIdentityService
+    {{
+        [{attributeNamespace}.OperationContract]
+        string Echo(params string[] inputs);
+    }}
```

Comment:
I didn't know you could use a params array with WCF!!

Reply:
I dontt know if it's supported at runtime either. If it is, the source gen will support it too.